py-polars/tests/unit/datatypes/test_temporal.py

```diff
@@ -562,7 +562,10 @@ def test_rolling() -> None:
         ("Europe/Warsaw", ZoneInfo("Europe/Warsaw")),
     ],
 )
-def test_upsample(time_zone: str | None, tzinfo: ZoneInfo | timezone | None) -> None:
+@pytest.mark.parametrize("offset", [None, "1mo"])
+def test_upsample(
+    time_zone: str | None, tzinfo: ZoneInfo | timezone | None, offset: None | str
+) -> None:
```

Comment:
I don't think you need to add this parametrisation to this test? There's already `test_upsample_crossing_dst` below which covers what happens if `offset` is passed to `upsample`

Reply:
The only reason we catch the warning in the `upsample_crossing_dst` test is that we use offset there, but that doesn't test the correctness of the output. Here we test for correctness of output, and I think from a general framework perspective it makes more sense to test that the deprecation of `upsample` keyword arguments happens in the `upsample` test, rather than a test dedicated to testing if DST is handled correctly. What are your thoughts?

Reply:
ok, sure